### DISCUSSION TITLE
Is there any more elegant way to fail a step other than raising exception

### DISCUSSION THREAD

**Discussion by samuelbenison** [getgauge/gauge#1915] [General]
Hi guys,

I have a question, not sure if it has been asked before, if so my apologies.

Is there any more elegant way to fail a step other than raising an exception (for example assertion error)?

When I assert fail a step, it throws an exception along with stack trace in the step, which is nice for some unexpected failures, but I would like to control that exception and return a more elegant way to fail the step maybe by adding a log message, etc for some steps.

Is it possible to do that? I tried to return 'False' for the gauge step implementation function but it is not failing the actual step.

Is there any way to achieve that? appreciate any and all suggestions.

Thanks

P.S: I am using gauge via Python.

**Comment by sriv** [comment]
In Gauge's view, a step would fail when it does not meet a desired outcome. The best way to ascertain that would be to use an assertion library, which can be set to throw contextual and crisp messages. Other than raising an exception, there's no way of telling Gauge to fail a step.

**Reply by samuelbenison** [reply]
So, is there a way to not print the stack trace in the HTML report for the step when raising an `AssertionError`?

I would like to keep the stack trace in the log file, not in my HTML report file.

For example, My html-report says the below for the failed step

```
Traceback (most recent call last):
   File "test.py", line 9, in <module>
      print KelvinToFahrenheit(-5)
   File "test.py", line 4, in KelvinToFahrenheit
      assert (Temperature >= 0),"Colder than absolute zero!"
AssertionError: Colder than absolute zero!
```

But I am trying to print only

```
AssertionError: Colder than absolute zero!
```

Can you please suggest how I can achieve that?

**Reply by sriv** [reply]
I could be missing something but that may not be possible without actually making gauge-python chomp the stacktrace, which will require a code change.
